Easydict/NewApp/View/SettingView/Tabs/DisabledAppTab.swift

```diff
@@ -0,0 +1,225 @@
+        EZLocalStorage.shared().selectTextTypeAppModelList = appModelList
+    }
+
+    func removeDisabledApp() {
+        appModelList = appModelList.filter { !selectedAppModels.contains($0) }
+        saveDisabledApps()
+        selectedAppModels = []
+    }
+
+    func newAppURLsSelected(from urls: [URL]) {
+        urls.forEach { url in
+            let gotAccess = url.startAccessingSecurityScopedResource()
+            if !gotAccess { return }
+            appendNewDisabledApp(for: url)
+            url.stopAccessingSecurityScopedResource()
+        }
+    }
+
+    func appendNewDisabledApp(for url: URL) {
+        guard let selectAppModel = disabledAppModel(from: url) else { return }
+        guard !appModelList.contains(selectAppModel) else { return }
+        appModelList.append(selectAppModel)
+        saveDisabledApps()
+    }
+
+    func disabledAppModel(from url: URL) -> EZAppModel? {
+        let appModel = EZAppModel()
+        guard let bundle = Bundle(url: url) else { return nil }
+        appModel.appBundleID = bundle.bundleIdentifier ?? ""
+        appModel.triggerType = []
+        return appModel
+    }
+}
+
+@available(macOS 13.0, *)
+struct DisabledAppTab: View {
+    @ObservedObject var disabledAppViewModel = DisabledAppViewModel()
+
+    var listToolbar: some View {
+        ListToolbar()
+            .fileImporter(
+                isPresented: $disabledAppViewModel.isImporting,
+                allowedContentTypes: [.application],
+                allowsMultipleSelection: true
+            ) { result in
+                switch result {
+                case let .success(urls):
+                    disabledAppViewModel.newAppURLsSelected(from: urls)
+                case let .failure(error):
+                    print("fileImporter error: \(error)")
+                }
+            }
+    }
+
+    var appListView: some View {
+        List(selection: $disabledAppViewModel.selectedAppModels) {
+            ForEach(disabledAppViewModel.appModelList, id: \.self) { app in
+                BlockAppItemView(with: app)
+                    .tag(app)
+            }
+            .listRowSeparator(.hidden)
+        }
+        .listStyle(.plain)
+        .scrollIndicators(.never)
+    }
+
+    var appListViewWithToolbar: some View {
+        VStack(spacing: 0) {
+            appListView
+
+            listToolbar
+        }
+        .clipShape(RoundedRectangle(cornerRadius: 8))
+        .padding(.bottom)
+        .padding(.horizontal, 35)
+        .onTapGesture {
+            disabledAppViewModel.selectedAppModels = []
+        }
+    }
+
+    var body: some View {
+        VStack {
+            Text("disabled_title")
+                .padding()
+
+            appListViewWithToolbar
+        }
+        .frame(maxWidth: 500)
+        .environmentObject(disabledAppViewModel)
+        .onAppear {
+            disabledAppViewModel.fetchDisabledApps()
+        }
+    }
+}
+
+@available(macOS 13.0, *)
+struct ListToolbar: View {
```

Comment:
Those views that is only used by `DisabledAppTab` should be marked as `private`.